### PR TITLE
Fixed compilation issues

### PR DESCRIPTION
**Pull request by lazerion**

### CODE REVIEW

File: demo/src/main/java/com/hazelcast/demo/AllTest.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.core.IMap;
 import com.hazelcast.internal.diagnostics.HealthMonitorLevel;
-import com.hazelcast.query.SqlPredicate;
+import com.hazelcast.query.impl.predicates.SqlPredicate;
```
Comment (vojtechtoman) [inline demo/src/main/java/com/hazelcast/demo/AllTest.java]: The idea is that users should not instantiate the predicates directly (using internal classes from `query.impl`). Instead, they should use the factory methods provided in `query.Predicates`. For the SQL predicate, there is `Predicates.sql()`, and there are also factory methods for the paging predicate, the partition predicate etc.

So I suggest to replace all direct instantiations of the impl classes with calls to the appropriate `Predicates` factory methods.

File: jcache/src/main/java/com/hazelcast/examples/splitbrain/AbstractCacheSplitBrainSample.java [hunk 2]
```diff
@@ -9,7 +9,7 @@
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.core.LifecycleEvent;
 import com.hazelcast.core.LifecycleListener;
-import com.hazelcast.instance.Node;
+import com.hazelcast.instance.impl.Node;
```
Comment (vojtechtoman) [inline jcache/src/main/java/com/hazelcast/examples/splitbrain/AbstractCacheSplitBrainSample.java]: Again, reference to `internal`/`impl` API

File: demo/src/main/java/com/hazelcast/demo/AllTest.java [hunk 3]
```diff
@@ -25,7 +25,7 @@
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.core.IMap;
 import com.hazelcast.internal.diagnostics.HealthMonitorLevel;
```
Comment (vojtechtoman) [inline demo/src/main/java/com/hazelcast/demo/AllTest.java]: Not sure if there is any other way, but do we really want our examples to depend on `impl`/`internal` packages?

File: helper/src/main/java/com/hazelcast/examples/helper/HazelcastUtils.java [hunk 4]
```diff
@@ -19,11 +19,11 @@
 import com.hazelcast.cluster.Cluster;
 import com.hazelcast.cluster.Member;
 import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.instance.impl.HazelcastInstanceImpl;
```
Comment (vojtechtoman) [inline helper/src/main/java/com/hazelcast/examples/helper/HazelcastUtils.java]: Again, reference to `internal`/`impl` API

File: spi/split-brain/src/main/java/ComposedHitsAndCreationTimeMapMergePolicyExample.java [hunk 5]
```diff
@@ -19,7 +19,7 @@
 import com.hazelcast.config.MapConfig;
 import com.hazelcast.config.MergePolicyConfig;
 import com.hazelcast.core.Hazelcast;
-import com.hazelcast.instance.HazelcastInstanceFactory;
+import com.hazelcast.instance.impl.HazelcastInstanceFactory;
```
Comment (vojtechtoman) [inline spi/split-brain/src/main/java/ComposedHitsAndCreationTimeMapMergePolicyExample.java]: Again, reference to `internal`/`impl` API

Comment (lazerion) [reply spi/split-brain/src/main/java/ComposedHitsAndCreationTimeMapMergePolicyExample.java]: Can you suggest alternative, I could not catch `internal` in package naming for `HazelcastInstanceFactory `

File: demo/src/main/java/com/hazelcast/demo/AllTest.java (hunk 1 above)
Comment (lazerion) [reply demo/src/main/java/com/hazelcast/demo/AllTest.java]: Done.

**Comment by lazerion** [comment]
@vojtechtoman PTAL

File: spi/split-brain/src/main/java/ComposedHitsAndCreationTimeMapMergePolicyExample.java (hunk 5 above)
Comment (lazerion) [reply spi/split-brain/src/main/java/ComposedHitsAndCreationTimeMapMergePolicyExample.java]: cc @Holmistr

Comment (vojtechtoman) [reply spi/split-brain/src/main/java/ComposedHitsAndCreationTimeMapMergePolicyExample.java]: Hi @lazerion, the correct API to use is `Hazelcast.newHazelcastInstance()`.

File: helper/src/main/java/com/hazelcast/examples/helper/HazelcastUtils.java (hunk 4 above)
Comment (lazerion) [reply helper/src/main/java/com/hazelcast/examples/helper/HazelcastUtils.java]: Done

**Comment by lazerion** [comment]
@Holmistr @vojtechtoman many thanks, I also addressed `impl` usages for `HazelcastInstanceFactory`

**Pull request merged**